---
title: "'Throw' 陳述式不可以在 'Catch' 陳述式外或 'Finally' 陳述式內省略運算元"
ms.date: 07/20/2015
f1_keywords:
- vbc30666
- bc30666
helpviewer_keywords:
- BC30666
ms.assetid: a208a6ea-0e36-4bf1-8984-4de1a0e38a2a
ms.openlocfilehash: c4ab115b01e78c245cbd9f564b50573d6e30c5e7
ms.sourcegitcommit: f8c270376ed905f6a8896ce0fe25b4f4b38ff498
ms.translationtype: MT
ms.contentlocale: zh-TW
ms.lasthandoff: 06/04/2020
ms.locfileid: "84414929"
---
# <a name="throw-statement-cannot-omit-operand-outside-a-catch-statement-or-inside-a-finally-statement"></a><span data-ttu-id="ac259-102">'Throw' 陳述式不可以在 'Catch' 陳述式外或 'Finally' 陳述式內省略運算元</span><span class="sxs-lookup"><span data-stu-id="ac259-102">'Throw' statement cannot omit operand outside a 'Catch' statement or inside a 'Finally' statement</span></span>
<span data-ttu-id="ac259-103">`Throw` 陳述式外的 `Catch` 陳述式必須提供例外狀況物件的名稱。</span><span class="sxs-lookup"><span data-stu-id="ac259-103">`Throw` statements outside of `Catch` statement must supply the name of an exception object.</span></span>  
  
 <span data-ttu-id="ac259-104">**錯誤 ID：** BC30666</span><span class="sxs-lookup"><span data-stu-id="ac259-104">**Error ID:** BC30666</span></span>  
  
## <a name="to-correct-this-error"></a><span data-ttu-id="ac259-105">更正這個錯誤</span><span class="sxs-lookup"><span data-stu-id="ac259-105">To correct this error</span></span>  
  
1. <span data-ttu-id="ac259-106">指定衍生自 `System.Exception`之例外狀況物件的名稱。</span><span class="sxs-lookup"><span data-stu-id="ac259-106">Specify the name of an exception object derived from `System.Exception`.</span></span>  
  
2. <span data-ttu-id="ac259-107">重建您的程式碼，讓 `Throw` 陳述式位於 `Catch` 區塊內。</span><span class="sxs-lookup"><span data-stu-id="ac259-107">Restructure your code so that the `Throw` statement is inside a `Catch` block.</span></span>  
  
## <a name="see-also"></a><span data-ttu-id="ac259-108">另請參閱</span><span class="sxs-lookup"><span data-stu-id="ac259-108">See also</span></span>

- [<span data-ttu-id="ac259-109">Throw 語句</span><span class="sxs-lookup"><span data-stu-id="ac259-109">Throw Statement</span></span>](../language-reference/statements/throw-statement.md)
- [<span data-ttu-id="ac259-110">Try...Catch...Finally 陳述式</span><span class="sxs-lookup"><span data-stu-id="ac259-110">Try...Catch...Finally Statement</span></span>](../language-reference/statements/try-catch-finally-statement.md)
- <xref:System.Exception?displayProperty=nameWithType>
- [<span data-ttu-id="ac259-111">在 .NET 中處理和擲回例外狀況</span><span class="sxs-lookup"><span data-stu-id="ac259-111">Handling and throwing exceptions in .NET</span></span>](../../standard/exceptions/index.md)
